packages/@aws-cdk/integ-tests/lib/assertions/providers/lambda-handler/assertion.ts

```diff
@@ -1,34 +1,63 @@
 /* eslint-disable no-console */
-import * as assert from 'assert';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { Match } from '@aws-cdk/assertions/lib/helpers-internal';
 import { CustomResourceHandler } from './base';
-import { AssertionRequest, AssertionResult } from './types';
+import { AssertionResult, AssertionRequest } from './types';
 
 export class AssertionHandler extends CustomResourceHandler<AssertionRequest, AssertionResult> {
   protected async processEvent(request: AssertionRequest): Promise<AssertionResult | undefined> {
+    let actual = request.actual;
+    const expected = decodeCall(request.expected);
     let result: AssertionResult;
+    let matcher;
+    console.log(`Testing equality between ${JSON.stringify(request.actual)} and ${JSON.stringify(request.expected)}`);
+
     switch (request.assertionType) {
-      case 'equals':
-        console.log(`Testing equality between ${JSON.stringify(request.actual)} and ${JSON.stringify(request.expected)}`);
-        try {
-          assert.deepStrictEqual(request.actual, request.expected);
-          result = { data: { status: 'pass' } };
-        } catch (e) {
-          if (e instanceof assert.AssertionError) {
-            result = {
-              data: {
-                status: 'fail',
-                message: e.message,
-              },
-            };
-          } else {
-            throw e;
-          }
+      case 'objectLike':
+        if (typeof actual === 'string') {
+          actual = JSON.parse(actual);
         }
+        matcher = Match.objectLike(expected);
+        break;
+      case 'equals':
+        matcher = Match.exact(expected);
+        break;
+      case 'arrayWith':
+        matcher = Match.arrayWith(expected);
         break;
       default:
         throw new Error(`Unsupported query type ${request.assertionType}`);
     }
 
+    const matchResult = matcher.test(actual);
+    matchResult.finished();
+    if (matchResult.hasFailed()) {
+      result = {
+        data: JSON.stringify({
+          status: 'fail',
+          message: [
+            ...matchResult.toHumanStrings(),
+            JSON.stringify(matchResult.target, undefined, 2),
+          ].join('\n'),
```

Comment:
If we choose to make the matcher succeed even if the assertion fails, do we also have a resource to make the deployment fail based on the outputs of all registered assertions? That will be necessary if we want to run assertions without the integ runner (like in CDK pipelines).

Reply:
We could just have a prop that the user could pass that would cause the custom resource to fail if the assertion fails.

Reply:
Added a prop, let me know what you think.